### PR TITLE
System messages

### PR DESCRIPTION
**Pull request by AClifton22**
Changed branch from alert to system messages

### CODE REVIEW

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export interface SystemMessageListProp {
+  type: SystemMessageNotificationTypes;
+  description: string;
+
+}
+
+const SystemMessageList  = () => {
+
+  const [BL, setBL] = useState<SystemMessageListProp[]>([]);
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Can you rename this to `[bannerList, setBannerList]`? I think that's more descriptive and easier to get what's going on.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, {useState} from 'react';
+import {SystemMessageNotification, SystemMessageNotificationTypes} from '../SystemMessageNotification/SystemMessageNotification';
+import {getSystemMessages} from '../../api';
+import {useMount} from '../../hooks/useMount';
+
+
+export interface SystemMessageListProp {
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Can you rename this interface? Something like `SysemMessage` will make it a little more clear what's going on here, I think.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  description: string;
+}
+
+export function SystemMessageNotification (props: Props) {
+
+    const [show] = useState(true);
+
+        return show ? (
+          <div className={styles['alert-container']}>
+             <Alert className={styles[props.type]} dismissible={true}  onClose={props.onClick}>
```
Comment (johnpcooke94) [inline src/components/SystemMessageNotification/SystemMessageNotification.tsx]: Your problem is here. Your API endpoint is serving the message type with a capitalized first letter, but your classnames in your CSS file are all lowercase.

You can fix this in two ways: either change the CSS file and `SystemMessageNotificationTypes` to use capital first letters, or change this line to use `className={styles[props.type.toLowerCase()]}`.

I don't know if @jasekiw has a preference but I think the former method is cleaner.

Comment (jasekiw) [reply src/components/SystemMessageNotification/SystemMessageNotification.tsx]: @AStiner @johnpcooke94 

I recommend not using classes dynamically. It is very hard to determine whether a class is being used or used correctly. Especially since each class is manually written.

I would create a function called getAlertClassName that takes the system message type and returns the fully qualified class name to use.

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+.info {
+    border-radius: 0 !important;
+    background: #3871dc;
+    color: white;
+    padding-bottom: 2px;
+    margin-bottom: 0px;
+   /* position: absolute;
+    right: 0;
+    left:0;*/
+    }
```
Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotification.module.scss:37]: @AStiner  You are duplicating a lot of style between these message types. Create a `.alert` class to hold all the base styles and use the remaining types as only modifications (ex. background color).

Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: @AStiner  You are duplicating a lot of style between these message types. Create a `.alert` class to hold all the base styles and use the remaining types as only modifications (ex. background color).

**Review by jasekiw — DISMISSED** [review]

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss (hunk 4 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: When I condensed this to the alert-container the border-radius would not work unless I included the specific type which removes the white space on the sides. I even tried to add it to Alert and it still didn't change it.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx (hunk 3 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.tsx]: I'm confused, you said that when you looked at the branch using this way was fine. When I tried to create a function like this you said it was unnecessary or was I going about it incorrectly before? Are you saying I need to create this function in list or notifications?  I went ahead and submitted what I was thinking.

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss (hunk 4 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: @jasekiw

Comment (jasekiw) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: How did you apply the class? Did you apply both classes (the base class and the alert type)? Having a different name doesn't change specificity so you shouldn't be running into issues there. The order that you declare these classes matter.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx (hunk 3 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.tsx]: @jasekiw

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss (hunk 4 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: @jasekiw Base class? Do you mean the alert-container? It calls the alert-container then the Alert I believe. But, I tried to add that to the alert-container or alert and it did not make a change. It does not look like the alert-container is called.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React, {useState} from 'react';
+import {Alert} from 'react-bootstrap';
+import styles from './SystemMessageNotification.module.scss';
+
+
+/*export function getSystemMessageType(type: Props) {
+  if (type.type === 'info') {
+    return 'info';
```
Comment (johnpcooke94) [inline src/components/SystemMessageNotification/SystemMessageNotification.tsx]: I think what Jason was after here was to return the CSS class itself, not a string containing the name of the class. So on this line you would return `styles['info']`, on line 11 you would return `styles['warning']`, line 14 return `styles['notice']`, etc.

And then on line 32, instead of passing the style directly into your component, you'd call this function and pass it the type name. So `className={getSystemMessageType(props.type.toLowerCase())}`

This way if you ever needed to add new banner types or remove existing ones, you would just update the function to reflect the new types.

Bear in mind I'm not at home so I can't compile this, the bits of code I wrote here might not be typesafe or work exactly as written, but hopefully you get the idea. @jasekiw feel free to let me know if I'm barking up the wrong tree here.

Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.tsx]: @johnpcooke94 Thanks, if that is what Jason was intending then it works.

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss (hunk 4 above)
Comment (AClifton22) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: @Figured it out. Let me know if this is what you are wanting.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React, {useState} from 'react';
+import {Alert} from 'react-bootstrap';
+import styles from './SystemMessageNotification.module.scss';
+
+
+export function getSystemMessageType(type: string) {
```
Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotification.tsx]: Map the original SystemMessageNotificationTypes over, not a string. This is susceptible to mistakes in the future.

File: src/components/SystemMessageNotification/SystemMessageNotification.module.scss (hunk 4 above)
Comment (jasekiw) [reply src/components/SystemMessageNotification/SystemMessageNotification.module.scss]: @AStiner This is not what I was wanting. You are using a sass feature when this is built into css.

Replace the `%` for `.` for the `alert-container` and remove all the `@extend` statements.

This however isn't a container, so it would be better called `.base-alert`.

In the component

Replace:

```
<Alert className={getSystemMessageType(props.type.toLowerCase())} dismissible={true}  onClose={props.onClick}>
```

with

```
<Alert className={styles['base-alert'] + ' ' + getSystemMessageType(props.type)} dismissible={true}  onClose={props.onClick}>
```

Now you've applied both the base class and the modifier class ex. `.info`.

Rename `getSystemMessageType` to `getAlertTypeClass`, change it's parameter to accept `SystemMessageNotificationTypes` not a string. Having it as a string opens up further mistakes in the future.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx (hunk 6 above)
Comment (AStiner) [reply src/components/SystemMessageNotification/SystemMessageNotification.tsx:6]: @jasekiw It throws an  error S2345: Argument of type 'string' is not assignable to parameter of type 'SystemMessageNotificationTypes'.

File: src/components/SystemMessageNotification/SystemMessageNotification.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react';
+import {Alert} from 'react-bootstrap';
+import styles from './SystemMessageNotification.module.scss';
+
+
+export function getAlertTypeClass(type: SystemMessageNotificationTypes) {
+  const newtype = type.toLowerCase();
```
Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotification.tsx]: @AStiner Why are you still mapping this to lowercase just to then check what the value is. This converts the type from `SystemMessageNotificationTypes` to a standard string, thus killing the type checking.

**Comment by AClifton22** [comment]
@jasekiw @johnpcooke94 Any advice on why the useEffect is not persisting through local storage when someone clicks the x button for the notification? It looks like it is not updating the count by 1.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const SystemMessagesList  = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [count, setCount] = useState(0);
+
+  useMount(async () => {
+      const messages = await getSystemMessages();
+      setSystemMessageList(messages);
+
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Here you'll want to check local storage for a count value, and set the `count` state accordingly. So something like this:

```
useMount(async () => {
      const messages = await getSystemMessages();
      setSystemMessageList(messages);
      const localCount = parseInt(window.sessionStorage.getItem('count'));
      if (localCount > 0) {
            setCount(localCount);
      }
});
```

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  // reference https://blog.bitsrc.io/5-methods-to-persisting-state-between-page-reloads-in-react-8fc9abd3fa2f
+  useEffect(()=> {
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Get rid of this useEffect call. We're setting the count state in the component onClick callback, so there's no need to do it here.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    setCount(JSON.parse(window.localStorage.getItem('count') as string));
+  }, []);
+
+  useEffect(() => {
+    window.localStorage.setItem('count', String(count));
+  }, ['count']);
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: This call might work as is, but a cleaner version would be something like
`useEffect(() => {
    window.sessionStorage.setItem('count', '${count}');
  }, [count]);`

If this doesn't work, try removing the `[count]`. I'm not sure if we need that dependency or not and I'm not at home to try it myself.

**Review by johnpcooke94 — CHANGES_REQUESTED** [review]
I don't think using local storage is a good idea. Local storage has no expiration, so if we use it  then the value will persist forever, and even if they close the page and come back on a new session, new messages won't be displayed. We should probably use sessionStorage instead, or even better store it in the Redux webstate.

As for why it's not working, I think your approach is flawed. Whatever function you pass into `useEffect` will run every time your component is rendered or re-rendered. So every time you change the `count` and trigger a re-render, you're both setting the `count` state to whatever you have in local storage, and then setting the local storage to whatever you have in the `count` state. This is a problem because you have both of those things happening in separate function calls, which execute in no particular order, but even if they did state updating in React is asynchronous, so now there's no guarantee on what value is even in the state when either of these execute.

You can read more about how the `useEffect` hook works [here](https://dmitripavlutin.com/react-useeffect-explanation/).

The flow you want should look something like this:
Component mounts -> check session storage for count value, if it's there load it into the component state, otherwise default to 0 -> user clicks an X -> component state updates -> update count value in session storage

I'll leave some code snippets on some file comments so you can hopefully see what I'm saying. Just bear in mind they might not be typesafe or work exactly as written, but hopefully you get the idea.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+const SystemMessagesList  = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [count, setCount] = useState( 0);
+
+  useMount(async () => {
+      const messages = await getSystemMessages();
+      setSystemMessageList(messages);
+
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Add a line here that will check for any IDs in session storage that have been dismissed. You'll need to track this in state. So something like this:

Replace line 20 with:
`const [dismissedMessages, setDismissedMessages] = useState<string[]>([]);`

Once you have that, on component mount you can check session storage for messages that have been displayed, and if there are any, add them to the state, like so:

```
const sessionDismissedMessages = window.sessionStorage.getItem('dismissedMessages').split(',');
if (sessionDismissedMessages) {
    setDismissedMessages(sessionDismissedMessages);
}
```

After this, if there are any message IDs stored in session storage, they will now be stored in the `dismissedMessages` state as an array of strings. Note that they *are* strings, so you will probably need to cast values to int or string, depending on usage, in most cases, but the TypeScript compiler will yell at you if you screw it up. The native js function `parseInt` works just fine for this.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  });
+
+     return (
+      <div>
+        {systemMessageList[count] ? <SystemMessageNotification onClick={() => setCount(count+1)} id={systemMessageList[count].id} type={systemMessageList[count].type} description={systemMessageList[count].description} />
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Here is where a big part of your approach will need to change. We have a list of system messages, and we have a list of IDs for messages that have already been dismissed. We need to iterate over those messages, and only display the ones that haven't been dismissed already. You can accomplish this with the `map()` method, which you can read about [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

```
{systemMessageList.map((m: SystemMessageList) => (
    !dismissedMessages.includes(m.id.toString()) ? <SystemMessageNotification onClick={() => setDismissedMessages([...dismissedMessages, m.id.toString()])} id={m.id} type={m.type} description={m.description} />
    : null
))}
```

This will display any messages that don't have an ID included in the `dismissedMessages` state, and when the message is clicked it will add the clicked message's ID to the state.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [count, setCount] = useState( 0);
+
+  useMount(async () => {
+      const messages = await getSystemMessages();
+      setSystemMessageList(messages);
+
+  });
+
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx:41]: You will need a `useEffect` callback here to update the IDs stored in session storage whenever the state updates. The state is an array of strings, and session storage always stores things as a plain old string, so you'll need to convert the string array to a string. Javascript has a method that can accomplish this called `join()`. The spec for this method is [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join).

Basically it just joins all the elements of an array into a string. By default it will separate the array entries by commas, which will work just fine for our purposes so we'll leave it alone. Like so:

```
useEffect(() => {
    window.sessionStorage.setItem('dismissedMessages', dismissedMessages.join());
}, [dismissedMessages]);
```

This should run every time the `dismissedMessages` state updates, and store the new value of that state in session storage under the key 'dismissedMessages'. We can then retrieve this state if the page ever reloads, and it will last until the browser session expires.

**Review by johnpcooke94 — DISMISSED** [review]
Your approach is going to have to change a bit compared to the last review I left. Your basic flow will generally stay the same:

Component mounts > check session storage for dismissed messages, if there are any load them into the component state > user clicks an X > component state updates > update dismissed messages in session storage

The big change is that we're tracking dismissed messages by ID now, so those are what you'll need to store in session storage, rather than a simple index of the number of messages. I've left some code snippets for you, hopefully they make sense. Again, might not work exactly as written, or be typesafe, but hopefully it'll get you started.

File: src/components/SystemMessageList/SystemMessageList.tsx (hunk 12 above)
Comment (AClifton22) [reply src/components/SystemMessageList/SystemMessageList.tsx]: @johnpcooke94 I tried this but I am receiving an  Expected an assignment or function call and instead saw an expression error. Have you had issues with something like this before? I'll commit my changes so far.

Comment (johnpcooke94) [reply src/components/SystemMessageList/SystemMessageList.tsx]: You might need to wrap this ternary in curly braces to signify that it's a javascript object. So like this:

```
{!dismissedMessages.includes(m.id.toString()) ? <SystemMessageNotification onClick={() => setDismissedMessages([...dismissedMessages, m.id.toString()])} id={m.id} type={m.type} description={m.description} /> : null}
```

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    const verifySessionID = window.sessionStorage.getItem('dismissedMessage').split(',');
+    if (verifySessionID) {
+      setDismissedMessage(verifySessionID);
+    }
+  });
+
+  return (
+    <div>
+      {systemMessageList.map((message: SystemMessageList) => {
+        if (!dismissedMessage.includes(message.id.toString())) {
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: @AStiner I see your problem. You can't use an if statement in a return expression like that. React is expecting jsx (basically just HTML with javascript objects in it) so you have to use a ternary operator, like I suggested.

Comment (AClifton22) [reply src/components/SystemMessageList/SystemMessageList.tsx]: Yep, figured it out thanks. Working on other error now with the windows session storage. TS2531: Object is possibly 'null'.

Comment (johnpcooke94) [reply src/components/SystemMessageList/SystemMessageList.tsx]: I think if you put a ! after `dismissedMessage` it might fix that:

`!dismissedMessage!.includes(...`

Comment (johnpcooke94) [reply src/components/SystemMessageList/SystemMessageList.tsx]: If not you'll probably have to extract this functionality into its own component

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    setSystemMessageList(messages);
+    const verifySessionID = window.localStorage.getItem ? ('dismissedMessage').split(',') : null;
+    if (verifySessionID) {
+      setDismissedMessage(verifySessionID);
+    }
+  });
+
+  return (
+    <div>
+      {systemMessageList.map((message: SystemMessageList) => {
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: This can get a lot simpler:

```
return (
    <div>
        <SystemMessageNotification 
        key={currentMessage.id} onClick{() => 
        setDismissedMessage([...dismissedMessage, currentMessage.id.toString()])}
        id={currentMessage.id}
        type={currentMesssage.type]
        description={currentMessage.description}
        />
    </div>
);
```

Now it will render whatever message is in the `currentMessage` state. We only need to make sure we update that state appropriately.

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+  id: number;
+  type: SystemMessageNotificationTypes;
+  description: string;
+
+}
+
+const SystemMessagesList = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: We will want a new state here to track whatever message we are currently displaying.

```
const [currentMessage, setCurrentMessage] = useState<SystemMessageList>();
```

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+    window.localStorage.setItem('dismissedMessage', dismissedMessage.join(','));
+  }, ['dismissedMessage']);
+
+  useMount(async () => {
+    const messages = await getSystemMessages();
+    setSystemMessageList(messages);
+    const verifySessionID = window.localStorage.getItem ? ('dismissedMessage').split(',') : null;
+    if (verifySessionID) {
+      setDismissedMessage(verifySessionID);
+    }
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx:39]: Once we have the dismissed messages from local storage, we need to search the messages and set the current active message to the first one we find that isn't dismissed. We can use [Array.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) for this.

```
setCurrentMessage(messages.find(m: SystemMessageList => !verifySessionID.includes(m.id.toString())));
```

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+const SystemMessagesList = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [dismissedMessage, setDismissedMessage] = useState<string[]>([]);
+
+  useEffect(() => {
+    window.localStorage.setItem('dismissedMessage', dismissedMessage.join(','));
```
Comment (johnpcooke94) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Add a line here that updates the `currentMessage` state:

```
setCurrentMessage(systemMessagesList.find(m: SystemMessageList => !dismissedMessage.includes(m.id.toString())));
```

**Review by johnpcooke94 — COMMENTED** [review]
We'll need to change our approach here a little bit. Since we're storing by IDs, and we only want to display one message at a time, we will need to search the messages we have, check them against dismissed messages, display the FIRST message we find that isn't already dismissed, and then stop. So our overall component flow will look like this:

Component mounts > check local storage for dismissed messages, if there are any load them into the component state > check the list of messages for any IDs that haven't been dismissed > display the first one we find

As usual I'll leave a few code snippets. Hopefully they help.

**Comment by Cliftonz** [comment]
@AStiner can you add a gif showing you implementation.

**Review by Cliftonz — COMMENTED** [review]
Lgtm, just add a gif of it working and I will approve it. Please show the local storage of the browser in the gif so we can see it change.

**Comment by AClifton22** [comment]
System-Message GIF

**Review by Cliftonz — APPROVED** [review]
Lgtm

File: src/components/SystemMessageList/SystemMessageList.tsx [hunk 19]
```diff
@@ -0,0 +1,60 @@
+import React, {useEffect, useState} from 'react';
+import {SystemMessageNotification, SystemMessageNotificationTypes} from '../SystemMessageNotification/SystemMessageNotification';
+import {getSystemMessages} from '../../api';
+import {useMount} from '../../hooks/useMount';
+
+export interface SystemMessageList {
```
Comment (jasekiw) [inline src/components/SystemMessageList/SystemMessageList.tsx]: Create a new file called `SystemMessage.ts` in the `src/types` folder and move this interface into it. Rename the interface to `SystemMessage` - It's not a list.

Comment (jasekiw) [reply src/components/SystemMessageList/SystemMessageList.tsx]: In webstorm you can have it automatically fix all usages by clicking on `SystemMessageList`, press Shift + F6 - it will highlight, backspace and type it out to be `SystemMessage` and press enter. That will fix all the usages to match the new name. Then click the name again and just press F6, this time it will prompt for the file to move it to, and make the tail end of the path match: `src\types\SystemMessage.ts` note: keep the C:\<folder to project> prefix in there. ex. 
image

**Review by jasekiw — CHANGES_REQUESTED** [review]

File: src/components/SystemMessage/SystemMessageCall.tsx [hunk 20]
```diff
@@ -0,0 +1,54 @@
+import React, {useEffect, useState} from 'react';
+import {SystemMessageNotif} from '../SystemMessageNotification/SystemMessageNotif';
+import {getSystemMessages} from '../../api';
+import {useMount} from '../../hooks/useMount';
+import {SystemMessage} from '../../types/SystemMessage';
+
+const SystemMessageCall = () => {
```
Comment (jasekiw) [inline src/components/SystemMessage/SystemMessageCall.tsx]: Why did you rename this? Rename this back to `SystemMessageList` and the filename too

File: src/components/SystemMessageNotification/SystemMessageNotif.tsx [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+    return styles['notice'];
+
+}
+
+interface Props {
+  onClick?: () => void;
+  id: number;
+  type: SystemMessageNotification;
+
+  description: string;
```
Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotif.tsx]: take these props out and replace it with the message object. Fix the caller and the definition
```suggestion
  message: SystemMessage
```

File: src/components/SystemMessageNotification/SystemMessageNotif.tsx [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export function SystemMessageNotif (props: Props) {
```
Comment (jasekiw) [inline src/components/SystemMessageNotification/SystemMessageNotif.tsx]: Rename this to `SystemMessageAlert` and the filename as well.

**Review by jasekiw — CHANGES_REQUESTED** [review]

**Review by jasekiw — APPROVED** [review]
